File: activityexample/src/main/java/com/academy/it/activityexample/MainActivity.java
package com.academy.it.activityexample;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private static final String KEY_NUMBER1 = "KEY_NUMBER1";
    private static final String KEY_NUMBER2 = "KEY_NUMBER2";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.startSecondActivity).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                intent.putExtra(KEY_NUMBER1, 999);
                intent.putExtra(KEY_NUMBER2, 3);
                startActivityForResult(intent, 1000);
            }
        });

        Log.d("MainActivity", "onCreate");
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if (requestCode == 1000 && resultCode == Activity.RESULT_OK && data != null) {
            Log.d("MainActivity", "Result " + data.getIntExtra("RESULT", 0));
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("MainActivity", "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("MainActivity", "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("MainActivity", "onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("MainActivity", "onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("MainActivity", "onDestroy");
    }
}

File: serviceexample/src/main/java/by/it/academy/serviceexample/MainActivity.kt
package by.it.academy.serviceexample

import android.content.ComponentName
import android.content.Intent
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private var serviceActions: ServiceActions? = null

    private val serviceConnection = object: ServiceConnection{
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            serviceActions = service as ServiceActions
            val data = serviceActions?.getData()
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            serviceActions = null
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.startService).setOnClickListener {
            startService(Intent(this@MainActivity, ServiceExample::class.java))
        }

        findViewById<Button>(R.id.stopService).setOnClickListener {
            stopService(Intent(this@MainActivity, ServiceExample::class.java))
        }

        bindService(
                Intent(this, ServiceExample::class.java),
                serviceConnection,
                0
        )
    }
}